File: backend/src/main/java/com/dietscheduler/backend/notification/NotificationSetting.java
package com.dietscheduler.backend.notification;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalTime;
import java.util.UUID;

@Entity
@Table(name = "notification_setting", uniqueConstraints = @UniqueConstraint(columnNames = "user_id"))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NotificationSetting {

    @Id
    @GeneratedValue
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "lunch_time", nullable = false)
    @Builder.Default
    private LocalTime lunchTime = LocalTime.of(12, 30);

    @Column(name = "dinner_time", nullable = false)
    @Builder.Default
    private LocalTime dinnerTime = LocalTime.of(19, 30);

    @Column(name = "snack_time", nullable = false)
    @Builder.Default
    private LocalTime snackTime = LocalTime.of(16, 0);

    @Column(name = "daily_meal_reminder_enabled", nullable = false)
    @Builder.Default
    private boolean dailyMealReminderEnabled = true;

    @Column(name = "start_cooking_reminder_enabled", nullable = false)
    @Builder.Default
    private boolean startCookingReminderEnabled = true;

    @Column(name = "low_stock_alert_enabled", nullable = false)
    @Builder.Default
    private boolean lowStockAlertEnabled = true;

    @Column(name = "low_stock_threshold", nullable = false)
    @Builder.Default
    private double lowStockThreshold = 1.0;
}

File: mobile/lib/models/notification_setting.dart
/// Times are "HH:mm:ss" strings as returned by the backend (java.time.LocalTime JSON form).
class NotificationSetting {
  final String lunchTime;
  final String dinnerTime;
  final String snackTime;
  final bool dailyMealReminderEnabled;
  final bool startCookingReminderEnabled;
  final bool lowStockAlertEnabled;
  final double lowStockThreshold;

  NotificationSetting({
    required this.lunchTime,
    required this.dinnerTime,
    required this.snackTime,
    required this.dailyMealReminderEnabled,
    required this.startCookingReminderEnabled,
    required this.lowStockAlertEnabled,
    required this.lowStockThreshold,
  });

  factory NotificationSetting.fromJson(Map<String, dynamic> json) => NotificationSetting(
        lunchTime: json['lunchTime'] as String,
        dinnerTime: json['dinnerTime'] as String,
        snackTime: json['snackTime'] as String,
        dailyMealReminderEnabled: json['dailyMealReminderEnabled'] as bool,
        startCookingReminderEnabled: json['startCookingReminderEnabled'] as bool,
        lowStockAlertEnabled: json['lowStockAlertEnabled'] as bool,
        lowStockThreshold: (json['lowStockThreshold'] as num).toDouble(),
      );

  String timeFor(String mealType) {
    switch (mealType) {
      case 'LUNCH':
        return lunchTime;
      case 'DINNER':
        return dinnerTime;
      case 'SNACK':
      default:
        return snackTime;
    }
  }
}
